### DISCUSSION TITLE
对比单体应用和微前端，有哪些问题是单体应用不用考虑而微前端需要考虑的？

### DISCUSSION THREAD

**Discussion by athrunsun** [umijs/qiankun#2767] [Q&A]
最近在我司实践qiankun的过程中，一直在思考是不是真正需要微前端的问题。对于标题中的问题，我自己想到了几点：

- qiankun自带的样式隔离功能不够强，还是需要额外的一些样式隔离处理，比如给使用不同版本antd的子应用加上不同的css class前缀
- 在子应用中加载第三方script (比如google maps) 可能会因为CORS的原因而失败
- 市面上绝大多数的library都没有考虑过微前端兼容性
- 本地开发时子应用失去HMR能力

还有什么其它的点呢？希望能从各位朋友那里集思广益一下，非常感谢！

**Comment by kuitos** [comment]
当你开始犹豫要不要微前端的时候，就说明你不需要微前端😂

**Comment by bravepg** [comment]
推荐阅读 @kuitos 的文章，https://zhuanlan.zhihu.com/p/391248835

另外提一下，微前端本质上关注的是如何解决组织和团队协作间带来的工程问题，而不是单纯的某个技术问题。因此，在落地微前端方案的时候，往往需要配套建设周边的基础设施，比如说：应用编排、监控、灰度等等。
